### DISCUSSION TITLE
Persistent Pending Session Requests on the Dapp sid

### DISCUSSION THREAD

**Discussion by hmendez-figure** [WalletConnect/WalletConnectFlutterV2#264] [Q&A]
Our team is currently developing a wallet with this lib (thanks BTW) and we are also using this lib to create a dApp for developing/testing against.

I have noticed that from the dApp side, when creating a request, it is not persisted like it is on the Wallet side. Is this by design? I can't seem to find in WC specs website on what should or shouldn't be persisted...

FWIW I did some digging and calling `Web3.request()` calls `core.pairing.sendRequest()` which eventually keeps track of pending dApp request in map in memory: `Map<int, PendingRequestResponse> pendingRequests = {}`. So if I restart the dApp and then try to respond from Wallet on that pending request, the dApp ignores it.

Any clarification would be great. Thanks!

**Comment by quetool** [comment]
Helo @hmendez-figure! Do you see this very same behavior when testing with WalletConnectFlutterV2's sample dApp and sample wallet? If so, do you mind sharing complete steps to reproduce the issue?

**Reply by hmendez-figure** [reply]
HI @quetool Thanks for the quick response! I tried this on the sample dApp & wallet.

<img width="1622" alt="image" src="https://github.com/WalletConnect/WalletConnectFlutterV2/assets/85638863/2709e741-f0a2-46b2-8853-69d52a247a36">

1. Connect dApp & Wallet (using `Sepolia` here)
2. Initiate request from dApp (using `personal_sign` here)
3. Restart dApp while waiting for response (screenshot)
4. `Reject` or `Approve` request from Wallet
5.  dApp ignores request because there is no record of it to complete

Not sure if this is the designed behavior or not. It would seem that if the Wallet is persisting the requests then the dApp should also be?
